src/main/java/com/joss/conductor/mobile/Locomotive.java

```diff
@@ -369,28 +342,104 @@ public String getAttribute(WebElement element, String attribute) {
+        return swipe(direction, getAppiumDriver().findElement(by), percentage);
+    }
+
+    public Locomotive swipe(SwipeElementDirection direction, WebElement element, float percentage) {
+        return swipe(direction, element, percentage, 0);
     }
 
     public Locomotive swipe(SwipeElementDirection direction, String id) {
         return swipe(direction, PageUtil.buildBy(configuration, id));
     }
 
     public Locomotive swipe(SwipeElementDirection direction, By by) {
-        return swipe(direction, by, SWIPE_DISTANCE);
+        return swipe(direction, getAppiumDriver().findElement(by));
     }
 
     public Locomotive swipe(SwipeElementDirection direction, WebElement element) {
-        return performSwipe(direction, element, /*by=*/null, SWIPE_DISTANCE);
+        return swipe(direction, element, SWIPE_DISTANCE, 0);
+    }
+
+    public Locomotive swipe(SwipeElementDirection direction, String id, int swipeDurationInMillis) {
+        return swipe(direction, PageUtil.buildBy(configuration, id), swipeDurationInMillis);
+    }
+
+    public Locomotive swipe(SwipeElementDirection direction, By by, int swipeDurationInMillis) {
+        return swipe(direction, getAppiumDriver().findElement(by), swipeDurationInMillis);
+    }
+
+    public Locomotive swipe(SwipeElementDirection direction, WebElement element, int swipeDurationInMillis) {
+        return swipe(direction, element, SWIPE_DISTANCE, swipeDurationInMillis);
     }
 
     public Locomotive swipeCenterLong(SwipeElementDirection direction) {
-        return performSwipe(direction, /*element=*/null, /*by=*/null, SWIPE_DISTANCE_LONG);
+        return swipe(direction, /*element=*/null, SWIPE_DISTANCE_LONG, 0);
+    }
+
+    public Locomotive swipeCenterLong(SwipeElementDirection direction, int swipeDurationInMillis) {
+        return swipe(direction, /*element=*/null, SWIPE_DISTANCE_LONG, swipeDurationInMillis);
+    }
+
+    public Locomotive swipeCenterLong(SwipeElementDirection direction, short numberOfSwipes) {
+        return swipeCenterLong(direction, 0, numberOfSwipes);
+    }
+
+    public Locomotive swipeCenterLong(SwipeElementDirection direction, int swipeDurationInMillis, short numberOfSwipes){
+        short i = 0;
+        while (i < numberOfSwipes) {
+            swipeCenterLong(direction, swipeDurationInMillis);
+            i++;
+        }
+        return this;
     }
 
     public Locomotive swipeCenterSuperLong(SwipeElementDirection direction) {
-        return performSwipe(direction, /*element=*/null, /*by=*/null, SWIPE_DISTANCE_SUPER_LONG);
+        return swipe(direction, /*element=*/null, SWIPE_DISTANCE_SUPER_LONG, 0);
+    }
+
+    public Locomotive swipeCenterSuperLong(SwipeElementDirection direction, int swipeDurationInMillis) {
+        return swipe(direction, /*element=*/null, SWIPE_DISTANCE_SUPER_LONG, swipeDurationInMillis);
+    }
+
+    public Locomotive swipeCenterSuperLong(SwipeElementDirection direction, short numberOfSwipes) {
```

Comment:
ditto about the overloads

Reply:
Good call. Removed the methods with a signature of only `short numberOfSwipes`.